x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java

```diff
@@ -229,14 +168,18 @@ public void afterBulk(long executionId, BulkRequest request, BulkResponse respon
                                 (msg1, msg2) -> Objects.equals(msg1, msg2) ? msg1 : msg1 + "," + msg2
                             )
                         );
-                    logger.error("failures: [{}]", failures);
+                    numFailed.addAndGet(failures.size());
+                    logger.error("Shard {} failed to populate rollup index: [{}]", indexShard.shardId(), failures);
                 }
             }
 
             @Override
             public void afterBulk(long executionId, BulkRequest request, Throwable failure) {
-                long items = request.numberOfActions();
-                numSent.addAndGet(-items);
+                if (failure != null) {
+                    long items = request.numberOfActions();
+                    numSent.addAndGet(-items);
```

Comment:
Even if it failed it was still "sent", right?

Reply:
I think here there may be a problem with retries. If failures are retried, then same docs are counted multiple times numSent. Right?

In the end we require that numSent == numIndexed, so if there are retries we may get failures because of this

Reply:
Fixed in [839a29d](https://github.com/elastic/elasticsearch/pull/85708/commits/839a29df872cfdae88af8c8885297e8811ced7b3)